src/esthri/tests/integration/upload_test.rs

```diff
@@ -16,6 +17,28 @@ fn test_upload() {
     assert!(res.is_ok());
 }
 
+#[test]
+fn test_upload_compressed() {
```

Comment:
nit. This compression functionality may be a decent candidate for a benchmark or two. I think out of scope for this PR though.

Reply:
Do you mean benchmark the speed of upload/download, or how many resources are used by the upload/download process?

Reply:
I meant the speed to more easily catch a regression of some sort. I'm not sure how heavily this will be in use but if we build infrastructure around it, a benchmark with upper/lower thresholds could make it easier to lazily keep tabs on it's functioning as intended.